APCAppCore/APCAppCore/Library/DataArchiverAndCollector/APCPassiveDataCollector.m

```diff
@@ -154,108 +151,20 @@ - (void) loadOrCreateDataFiles: (APCDataTracker*) tracker
-    
-    [self createZipFile:tracker];
-    [self resetDataFilesForTracker:tracker];
-}
 
-- (void) createZipFile:(APCDataTracker*) tracker
-{
-    NSError * error;
-    NSString * unencryptedZipFileName = [NSString stringWithFormat:@"unencrypted_%@_%0.0f.zip",tracker.identifier, [[NSDate date] timeIntervalSinceReferenceDate]];
-    NSString * encryptedZipFileName = [NSString stringWithFormat:@"encrypted_%@_%0.0f.zip",tracker.identifier, [[NSDate date] timeIntervalSinceReferenceDate]];
-    NSString * unencryptedPath = [self.collectorsUploadPath stringByAppendingPathComponent:unencryptedZipFileName];
-    NSString * encryptedPath = [self.collectorsUploadPath stringByAppendingPathComponent:encryptedZipFileName];
+    NSError *flushError = nil;
     
-    ZZArchive * zipArchive = [[ZZArchive alloc] initWithURL:[NSURL fileURLWithPath:unencryptedPath]
-                                                    options:@{ZZOpenOptionsCreateIfMissingKey : @YES}
-                                                      error:&error];
-    APCLogError2(error);
-    NSMutableArray * zipEntries = [NSMutableArray array];
-    NSString * csvFilePath = [tracker.folder stringByAppendingPathComponent:kCSVFilename];
-    NSString * infoFilePath = [tracker.folder stringByAppendingPathComponent:kInfoFilename];
-
-    APCLogFilenameBeingArchived (kCSVFilename);
-    APCLogFilenameBeingArchived (kInfoFilename);
-
-    [zipEntries addObject: [ZZArchiveEntry archiveEntryWithFileName: kCSVFilename
-                                                           compress:YES
-                                                          dataBlock:^(NSError** __unused error){ return [NSData dataWithContentsOfFile:csvFilePath];}]];
-    [zipEntries addObject: [ZZArchiveEntry archiveEntryWithFileName: kInfoFilename
-                                                           compress:YES
-                                                          dataBlock:^(NSError** __unused error){ return [NSData dataWithContentsOfFile:infoFilePath];}]];
+    BOOL successfullyMoved = [APCDataArchiverAndUploader uploadFileAtPath:dataFilePath
+                                                       withTaskIdentifier:tracker.identifier
+                                                           andTaskRunUuid:[NSUUID UUID]
+                                                           returningError:&flushError];
     
-    [zipArchive updateEntries:zipEntries error:&error];
-    APCLogError2(error);
-    
-    [APCDataArchiver encryptZipFile:unencryptedPath encryptedPath:encryptedPath];
-    APCLogDebug(@"Created zip file: %@", encryptedPath);
-    
-#ifdef USE_DATA_VERIFICATION_CLIENT
-    [APCDataVerificationClient uploadDataFromFileAtPath: unencryptedPath];
-#else
-    NSError * deleteError;
-    if (![[NSFileManager defaultManager] removeItemAtPath:unencryptedPath error:&deleteError]) {
-        APCLogError2(deleteError);
+    if (!successfullyMoved) {
+        APCLogError2(flushError);
```

Comment:
Haven't seen it fail yet, but when it does, the only thing is to not touch the data file. That way the collector will keep appending to it; and the upload will be tried again.